### ISSUE TITLE
Create a Fuel block explorer backend

### ISSUE THREAD

**Issue by ra0x3** [FuelLabs/fuel-indexer#130]
- Create a production-ready Fuel Block Explorer backend using the Fuel Indexer service
- This service should read blocks & transactions from a Fuel node and persist them into a database
- This Block Explorer would run as a single service, reading a single manifest
- This isn't really blocked by https://github.com/FuelLabs/fuel-indexer/issues/104 -- but once https://github.com/FuelLabs/fuel-indexer/issues/104 lands, this example (the block explorer) would obviously have to be updated accordingly

Additional context
- WASM execution only
- We could potentially keep this project in `exampes/block-explorer` in order to allow people to easily see how to replicate the functionality
- Service would need some way to sync state
  - If for some reason the service goes offline, it would need to know where to pick up where it left off
 (so on and so forth)
- Service would be hosted by Fuel
- Would require an entry into the mdbook
- Would possibly be needed to be built into its own binary and published using CI/CD
- Users would be able to search by address, TX hash, block, token, or ENS 
  - Lots of the same functionality as https://etherscan.io/
- This might be a good opportunity to play with schema versioning
  - E.g., We run the service with some schema so the database has some state (after it starts indexing items), then we update the schema, generate a new WASM blob, create a new manifest, then somehow upload these new assets to the service with no downtime
    - This might be potentially out of scope (scope creep!), but just pointing out that this is functionality we plan to add, so might be good to implement this now since the actual block explorer service piece is so relatively straightforward

**Comment by ra0x3** [comment]
CC @mitchmindtree @JoshuaBatty @tjsharp1

**Comment by ra0x3** [comment]
Closing, as we've broken this down into smaller issues
